### PR TITLE
[ci] Split jobs and fix deployments

### PR DESCRIPTION
**Pull request by alexandrebouthinon**
## What does this PR do ?
Fix deployments (S3 and Github release)
Split Travis CI builds

### How should this be manually tested?
Check Travis and/or try to reproduce builds

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,66 +1,170 @@
...
+        - npm run unit
+
+      after_script:
+        - npm run codecov
+
+# ---------------------------------------
+# Functionnal Testing
+# ---------------------------------------
+    - stage: tests
+      name: "Functionnal Tests"
```
Comment (xbill82) [inline .travis.yml]:
```suggestion
      name: "End to end Tests"
```

Comment (alexandrebouthinon) [reply .travis.yml]: Yep, I'll fix that :slightly_smiling_face:

File: .ci/package_electron.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# Print help if no arguments passed
+if [[ "$#" -eq "0" ]]; then
+  echo "No arguments supplied, please pass packaging type and plateform."
+  echo "Example: ./package_electron.sh compile win32"
+  echo "Example: ./package_electron.sh package linux"
+  echo "NOTE: Available plateform are win32, darwin and linux"
+  exit 1
+fi
+
+PACKAGE_VERSION=$(cat package.json | grep version | head -1 | awk -F: '{ print $2 }' | sed 's/[\",]//g' | tr -d '[[:space:]]')
```
[the remark this answers is not in the document]
Comment (alexandrebouthinon) [reply .ci/package_electron.sh]: I agree with you but this is another dependency to install in Travis builder, and we can do the same using bash

File: .ci/release_package.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+set -e
+
+PACKAGE_VERSION=$(cat package.json | grep version | head -1 | awk -F: '{ print $2 }' | sed 's/[\",]//g' | tr -d '[[:space:]]')
```
[the remark this answers is not in the document]
Comment (alexandrebouthinon) [reply .ci/release_package.sh]: Same as above

[the remark this answers is not in the document]
Comment (etrousset) [reply .ci/release_package.sh]: Ok, we can wait the day we need to troubleshot this command line to make the switch

**Pull request merged**